src/main/scala/com/ing/wbaa/gargoyle/proxy/provider/LineageProviderAtlas.scala

```
@@ -0,0 +1,143 @@
+    for {
+      entityGuid <- getEntityGUID(typeName, entityName)
+      _ <- deleteEntity(entityGuid)
+
+    } yield (Some(LineagePostGuidResponse("", "", entityGuid, "")))
+  }
+
+  def createLineageFromRequest(httpRequest: HttpRequest, userSTS: User): Future[Option[LineagePostGuidResponse]] = {
+
+    val timestamp = System.currentTimeMillis()
+    // createLineageFromRequest is used in RequestHandlerS3 where there is no notion of S3Request. It seems easier to
+    // repeat httpRequest parsing here for needed values
+    val host = httpRequest.uri.authority.host.address()
+    val path = httpRequest.uri.path
+    val bucket = path.toString.split("/").toList.lift(1).getOrElse("notDef")
+    val bucketObject = s"${path.toString.split("/").toList.lift(2).getOrElse("emptyObject")}"
+    val method = httpRequest.method
+    val contentType = httpRequest.entity.contentType
+    val userName = userSTS.userName.value
+
+    if (bucket != "notDef" && bucketObject != "emptyObject") {
+      logger.debug(s"Creating lineage for request ${method.value} file ${bucketObject} in ${bucket} at ${timestamp}")
+      method match {
+        case HttpMethods.GET =>
+          logger.debug(s"Creating Read lineage for request to ${method.value} file ${bucketObject} to ${bucket} at ${timestamp}")
+          postEnities(userName, host, bucket, bucketObject, "read", contentType, timestamp)
+        case HttpMethods.POST | HttpMethods.PUT =>
+          logger.debug(s"Creating Write lineage for request to ${method.value} file ${bucketObject} to ${bucket} at ${timestamp}")
+          postEnities(userName, host, bucket, bucketObject, "write", contentType, timestamp)
+        case HttpMethods.DELETE =>
+          logger.debug(s"Creating Delete lineage for request to ${method} file ${bucketObject} to ${bucket} at ${timestamp}")
+          deleteEntities("DataFile", bucketObject)
+        case _ => Future(None)
+      }
+    } else {
+      Future(None)
```

instead of a Future[Option[...]] we could also return just a Future[...]. Here you could let the future fail since in this case it is actually a failed attempt to write lineage

---

yes, changed to feature failed